pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -188,16 +181,14 @@ func (manager *gceServiceManager) AttachDiskOnCloudProvider(
 		Source:     source,
 		Type:       diskTypePersistent,
 	}
-	return manager.gce.service.Instances.AttachDisk(
-		manager.gce.projectID, instanceZone, instanceName, attachedDiskV1).Do()
+	return manager.gce.c.Instances().AttachDisk(context.Background(), meta.ZonalKey(instanceName, instanceZone), attachedDiskV1)
```

Comment:
is Background() the right context? Will this retry and timeout like the old WaitForZoneOp? Ditto everywhere else

Reply:
Background() is used in instance calls as well. It doesn't seem like there's a retry mechanism in place currently - @bowei should there be?

Reply:
Background() is used in instance calls as well, and it doesn't seem like context is used for retry and timeouts